modules/caddyhttp/encode/zstd/zstd.go

```diff
@@ -39,6 +44,27 @@ func (Zstd) CaddyModule() caddy.ModuleInfo {
 
 // UnmarshalCaddyfile sets up the handler from Caddyfile tokens.
 func (z *Zstd) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	d.Next() // consume option name
+	if !d.NextArg() {
+		return nil
+	}
+	levelStr := d.Val()
+	if ok, _ := zstd.EncoderLevelFromString(levelStr); !ok {
+		return d.Errf("unexpected compression level, use one of '%s', '%s', '%s', '%s'",
+			zstd.SpeedFastest,
+			zstd.SpeedDefault,
+			zstd.SpeedBetterCompression,
+			zstd.SpeedBestCompression,
+		)
+	}
+
+	z.Level = levelStr
+	return nil
+}
+
+// Provision provisions z's configuration.
+func (z *Zstd) Provision(ctx caddy.Context) error {
+	_, z.level = zstd.EncoderLevelFromString(z.Level)
```

Comment:
We actually need to check if the level is correct here. Power users will use JSON to configure caddy plugins.

We need to check if the level string is valid in 2 places, `UnmarshalCaddyfile` and `Provision`, or `Validate` if you prefer to implementing another interface.

Also there needs to be a default value in case it's not set.

Reply:
Made a quick commit to do this